Framework/Algorithms/inc/MantidAlgorithms/CalculateCarpenterSampleCorrection.h

```diff
@@ -0,0 +1,101 @@
+#ifndef MANTID_ALGORITHM_MULTIPLE_SCATTERING_ABSORPTION_H_
+#define MANTID_ALGORITHM_MULTIPLE_SCATTERING_ABSORPTION_H_
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/DistributedAlgorithm.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidHistogramData/Points.h"
+#include <vector>
+
+namespace Mantid {
+namespace HistogramData {
+class HistogramX;
+class HistogramY;
+class HistogramE;
+}
+namespace Algorithms {
+/** Multiple scattering absorption correction, originally used to
+    correct vanadium spectrum at IPNS.  Algorithm originally worked
+    out by Jack Carpenter and Asfia Huq and implmented in Java by
+    Alok Chatterjee.  Translated to C++ by Dennis Mikkelson.
+
+    @author Dennis Mikkelson
```

Comment:
Minor detail: you can remove the `@author` and `@date`. Also, please update the copyright line.